lib/galaxy/datatypes/microarrays.py

```diff
@@ -0,0 +1,152 @@
+import logging
+import subprocess
+
+from galaxy.datatypes import data
+from galaxy.datatypes.metadata import MetadataElement
+
+log = logging.getLogger(__name__)
+
+
+def get_n_first_lines(filename, n=3):
+    """
+        Rendering n first lines in order to populate peek.
+    """
+    try:
+        cmd = ["head", '-' + str(n)]
+        cmd.extend([filename])
+        out = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+        return out.communicate()[0]
+    except Exception:
+        pass
+    return 0
```

Comment:
I think you can remove this function and use `get_file_peek` , which can be imported as:
```
from galaxy.datatypes.data import get_file_peek
```